_posts/2023/2023-07-07-cobbler-inside-libvirt-1.md

```diff
@@ -0,0 +1,300 @@
+nmcli con mod "Wired connection 1" ipv4.method "manual"
+nmcli c up "Wired connection 1"
+```
+
+> You will need to adjust the name of the interface `Wired Connection 1` to the interface that isn't online. This will
+> be the interface written in white normally. The other two interfaces should have a green tone.
+
+### Install & Configure Cobbler
+
+At the time of writing this guide, the current version of Cobbler in openSUSE Tumbleweed is 3.3.3. This can be 
+installed with:
+
+```shell
+zypper in cobbler
+```
+
+Confirm the installation with `y` when prompted and confirm by hitting the enter key. This will also install the TFTP 
+server along with several other required dependencies.
+
+In a default openSUSE Tumbleweed installation `firewalld` is configured and active. To open the required ports we 
+execute the following commands:
+
+```shell
+firewall-cmd --permanent --zone=public --add-service=http
+firewall-cmd --permanent --zone=public --add-service=https
+firewall-cmd --permanent --zone=public --add-service=tftp
+firewall-cmd --permanent --zone=public --add-service=dhcp
+firewall-cmd --permanent --zone=public --add-service=dhcpv6
+```
+
+> Be sure that this is executed on the guest. Opening these ports on the host may have unexpected consequences.
+
+Since now clients can access the desired services we should start them up.
+
+- Enable the TFTP service with: `systemctl enable --now tftp.service`
+- Enable the webserver with: `systemctl enable --now apache2.service`
+- Enable the DHCP service with: `systemctl enable --now dhcpd.service` - this will fail but this is expected at this 
+  point
+
+Now we can finally start configuring Cobbler itself. To do so edit the central configuration file at 
+`/etc/cobbler/settings.yaml` and make it match the IP on the guest (ignore the rest of the settings file for now):
+
+```yml
+server: 10.17.3.2
+next_server_v4: 10.17.3.2
+manage_dhcp: true
+manage_dhcp_v4: true
+```
+
+For this to take effect restart Cobbler:
+
+```shell
+systemctl restart cobblerd
+```
+
+Since Cobbler now knows its own IP address we can modify the DHCP template that Cobbler ships to match our requirements.
+To do so adjust the file `/etc/cobbler/dhcp.template`. The important part of the file should look like this after 
+modifications:
+
+```
+subnet 10.17.3.0 netmask 255.255.255.0 {
+        option routers             10.17.3.1;
+        option domain-name-servers 8.8.8.8;
+        option subnet-mask         255.255.255.0;
+        range dynamic-bootp        10.17.3.100 10.17.3.254;
+```
+
+The rest of the template can be left as is. Cobbler per-default uses [Cheetah](https://cheetahtemplate.org/) as a 
+template language. It supports [Jinja](https://jinja.palletsprojects.com/en/3.1.x/) via a shebang as well.
+
+## Importing your first distribution to Cobbler
+
+Now at this point Cobbler doesn't do much. It doesn't hand out DHCP addresses since we haven't defined a DHCP pool, the 
```

Comment:
punctuation
```suggestion
Now, at this point Cobbler doesn't do much. It doesn't hand out DHCP addresses since we haven't defined a DHCP pool, the 
```